### ISSUE TITLE
Dimension labels are strange after scaled cut_md

### ISSUE THREAD

**Issue by NickDraper** [mantidproject/mantid#12582]
This issue was originally [TRAC 11744](http://trac.mantidproject.org/mantid/ticket/11744)

Run the following script look at cut_r, the scaling although correct (bragg peaks in integer positions), has a side effect of altering the dimension axes to:
* ['1.11zeta',0,0]
* [0, '1.11eta', 0]
* [0, 0, '1.11x1']

Investigate if there is anything we can do here

```python
ws = Load(Filename='SXD23767.raw',OutputWorkspace='SXD23767',LoadMonitors='Exclude')

SetUB(ws, UB=[-0.15097235,  0.09164432 , 0.00519473 ,0.0831895,   0.14123681, -0.06719047, -0.0384503, -0.05534039, -0.1633801 ])
s = ws.sample()
SetGoniometer(ws)

scales =  'Q in A^-1' # 'Q in lattice units' 
HKL = ConvertToMD(InputWorkspace=ws, QConversionScales=scales, QDimensions='Q3D', dEAnalysisMode='Elastic', Q3DFrames='HKL', OutputWorkspace='HKL', MinValues='-10,-10,-10', MaxValues='10,10,10')


HKL = mtd['HKL']
p = Projection()
p.setType(0, 'r')
p.setType(1, 'r')
p.setType(2, 'r')
proj = p.createWorkspace('proj')

cut_r = CutMD(HKL, Projection=proj, NoPix=True, PBins=[[0.1], [0.1], [0.1]])

p = Projection()
p.setType(0, 'a')
p.setType(1, 'a')
p.setType(2, 'a')
proj = p.createWorkspace('proj')

cut_a = CutMD(HKL, Projection=proj, NoPix=True, PBins=[[0.1], [0.1], [0.1]])

l = -4.0
svw1 = plotSlice(cut_r, colorscalelog=True)
svw1.setSlicePoint(2, l)
svw2 = plotSlice(cut_a, colorscalelog=True)
svw2.setSlicePoint(2, l)
```

**Comment by matthew-d-jones** [comment]
I do not get stated behaviour, for cut_r the labels are:
['zeta', 0, 0]
[0, 'eta', 0]
[0, 0, 'xi']
Last change to relevant code was 2 months before the ticket was opened, so not sure what has happened here. Do we know if this is now the desired behaviour?

**Comment by AndreiSavici** [comment]
The labels should match the ones from ConvertToMD (zeta, eta, xi might be used instead of H,K,L).
The cut_a should be `['zeta',0,0](A-1)`, matching dimensions for KHL when scales are in `Q in A^-1`
The cut_r should be `['zeta',0,0](in 1.11A-1)`, matching dimensions for KHL when scales are in `HKL`

**Comment by matthew-d-jones** [comment]
Commit cfc08e2 fixes #15743, so now CutMD unit handling is only as broken as this ticket suggests.

**Comment by matthew-d-jones** [comment]
Commit  3733e73 fixes #15761.
